soccer/gameplay/tactics/wall.py

```diff
@@ -0,0 +1,65 @@
+import composite_behavior
+import behavior
+import constants
+import robocup
+import main
+import enum
+import math
+import skills.move
+
+# This tactic builds a wall a certain distance from point A, blocking point B.
+class Wall(composite_behavior.CompositeBehavior):
+    class State(enum.Enum):
+        defense_wall = 1
+
+    def __init__(self,
+                 num_defenders = 3,                         # number of defenders we're making the wall with (default 3)
+                 curvature = 0,                             # curvature (in radians) of the wall 0 - pi/2
+                 mark_point = None,                         # what point we are defending against (default is ball)
+                 defender_point = robocup.Point(0, 0),      # what point we are defending (default is goal)
+                 defender_spacing = 3,                      # number of robot radii between the centers of the defenders in the wall
+                 dist_from_mark = 1):                       # distance from the mark point we want to build the wall
+        super().__init__(continuous=True)
+
+        self.number_of_defenders = num_defenders
+        self.curvature = curvature
+        self.mark_point = main.ball().pos if mark_point == None else mark_point
```

Comment:
Ball can be None if its outside vision

Reply:
How should we handle that?

Reply:
Sorry, that's wrong on my end. Ball will always have a value, but it may be an invalid (`main.ball().valid = false`) ball. There are two common cases for an invalid ball, the first is if someone grabs the ball and literally leaves vision. The position will be the last known location. The second common case is when a robot blocks the view of the camera from seeing the ball. The ball will still be in play and actionable. 

Just due to the second case, it's probably a better idea to keep it like this and just always use the latest ball position